vyper/venom/venom_to_assembly.py

```diff
@@ -295,19 +295,12 @@ def _generate_evm_for_basicblock_r(
 
         self.clean_stack_from_cfg_in(asm, basicblock, stack)
 
-        param_insts = [inst for inst in basicblock.instructions if inst.opcode == "param"]
-        main_insts = [inst for inst in basicblock.instructions if inst.opcode != "param"]
+        param_insts = (inst for inst in basicblock.instructions if inst.opcode == "param")
+        main_insts = (inst for inst in basicblock.instructions if inst.opcode != "param")
+        all_inst = [x for one_iter in (param_insts, main_insts) for x in one_iter]
```

Comment:
```suggestion
        all_insts = param_insts + main_insts
```

Reply:
This would work if the `param_insts` and `main_insts` would be arrays and not generators, the reason I made them generators is so I dont have to allocate more lists, but for sure I could change that